ui/src/components/DetectionsTable.tsx

```diff
@@ -16,18 +24,26 @@ import { DetectionsPlayer } from "./Player/DetectionsPlayer";
 export default function DetectionsTable({
   detections,
   feed,
-  candidate
+  candidate,
+  onDetectionUpdate,
 }: {
   detections: Detection[];
   feed: Pick<Feed, "slug" | "nodeName">;
-  candidate: Pick<Candidate, "id">;
+  candidate: Pick<Candidate, "id" | "visible">;
+  onDetectionUpdate: () => void;
 }) {
   const offsetPadding = 15;
   const minOffset = Math.min(...detections.map((d) => +d.playerOffset));
   const maxOffset = Math.max(...detections.map((d) => +d.playerOffset));
   const startOffset = Math.max(0, minOffset - offsetPadding);
   const endOffset = maxOffset + offsetPadding;
 
+  const { currentUser } = useGetCurrentUserQuery().data ?? {};
+
+  const setDetectionVisible = useSetDetectionVisibleMutation({
+    onSuccess: onDetectionUpdate,
```

Comment:
What do you think about using a [query invalidation](https://tanstack.com/query/v4/docs/react/guides/invalidations-from-mutations) instead of sending the action up for a refetch? Something like this:

```suggestion
    onSuccess: () => {
      queryClient.invalidateQueries(
        useCandidateQuery.getKey({ id: candidate.id }),
      );
    },
```

It avoids having to add a callback, though it does add more coupling. I feel like that's fine since the DetectionsTable is already highly coupled?

(also not a huge concern at the moment but want to get these patterns right if we're gonna be copying them around in the future)

Reply:
Ah nice, I didn't know about this one. I can get this changed.